src/test/java/uk/org/webcompere/modelassert/json/ExamplesTest.java

```diff
@@ -541,4 +541,14 @@ void atPath_whenKeysAreMissingInTheActualThenObjectContainsFixesIt() {
             .path("z").objectContains()
             .isEqualTo("{z:{b:{d:false, c:true}}}");
     }
+
+    @Test
+    void at_isCheckedEvenIfTreeComparisonSucceeds() {
+        assertThatThrownBy(() ->
+                assertJson("{foo: 42}")
+                        .where().objectContains()
+                        .at("/foo").isText()
+                        .isEqualTo("{}"))
+                .isInstanceOf(AssertionFailedError.class);
```

Comment:
I can explain this one. It's sort of user-error, but it's sort of not.

When we use `objectContains` we're essentially relaxing the rules. We're telling the asserter to ignore any field not present in the _expected_.

The following version of your test fails:

```java
assertThatThrownBy(() ->
        assertJson("{foo: 42}")
                .where().objectContains()
                .at("/foo").isText()
                .isEqualTo("{foo:42}"))
        .isInstanceOf(AssertionFailedError.class);
```